Add ConditionKernelCommandline to cloud-config/cloud-final

Otherwise cloud-config/cloud-final might start without cloud-init
enabled first (e.g. during install).

---

--- a/systemd/cloud-config.service.tmpl
+++ b/systemd/cloud-config.service.tmpl
@@ -4,7 +4,7 @@
 After=network-online.target cloud-config.target configd-loaded.target
 After=snapd.seeded.service
 Wants=network-online.target cloud-config.target
-ConditionFileNotEmpty=/var/lib/cloud/instance/datasource
+ConditionKernelCommandLine=cloud-init
 
 [Service]
 Type=oneshot
--- a/systemd/cloud-final.service.tmpl
+++ b/systemd/cloud-final.service.tmpl
@@ -7,7 +7,7 @@
 Before=apt-daily.service
 {% endif %}
 Wants=network-online.target cloud-config.service
-ConditionFileNotEmpty=/var/lib/cloud/instance/datasource
+ConditionKernelCommandLine=cloud-init
 
 
 [Service]
--- a/systemd/cloud-init.service.tmpl
+++ b/systemd/cloud-init.service.tmpl
@@ -36,8 +36,6 @@
 ConditionKernelCommandLine=cloud-init
 # Triggered by IP auto-configuration
 ConditionPathExistsGlob=|/run/net-*.conf
-# Triggered by cloud-init-local
-ConditionFileNotEmpty=|/var/lib/cloud/instance/datasource
 
 [Service]
 Type=oneshot
